### ISSUE TITLE
Ipywidgets.interact is not working on datahub

### ISSUE THREAD

**Issue by rohanjha123** [berkeley-dsep-infra/hubs#4067]
### Bug description

While other functions in the Ipywidgets package are working fine, the interact() function is no longer working on datahub. I have old notebooks where interact() used to work and it no longer works. As I haven't made any edits to the notebook since March 2021, I believe this is an issue with the general datahub and not with any changes I might have made. A concise description of what interact used to do and what it does now is written below:

interact() would take in a function and a list of different values for the function parameters. It would determine an appropriate widget to use with the function parameters and allow users to see how the function output changed with different values. Now, interact merely generates the function output with the first possible value out of the list, and it doesn't generate a widget allowing the user to vary the function parameter.

If it helps, my bug is similar to the one described here: https://github.com/jupyter-widgets/ipywidgets/issues/1778. 

### Environment & setup

- Hub: Datahub
- Language: Python

### How to reproduce

Try running the following code in any datahub notebook, or I can send a path to a notebook with the bug on the UCTech Slack.
```
import ipywidgets as widgets
from ipywidgets import interact, interactive, fixed, interact_manual
def f(x):
    return 3 * x
interact(f, x=10);
```

**Comment by rohanjha123** [comment]
I have tried the debugging steps linked in the GitHub thread, as well as running the notebook in both chrome and safari

**Comment by felder** [comment]
When I look at the developer tools in Firefox I see the following:

![Screen Shot 2023-01-05 at 9 57 27 AM](https://user-images.githubusercontent.com/1211730/210824419-988fbd8c-4996-48ed-946f-83f1d81b1e59.png)

Perhaps related?

**Comment by felder** [comment]
I upgraded ipywidgets to the latest version on my server and reloaded the kernel.  The error above disappeared, but the output was unchanged.

**Comment by balajialg** [comment]
@rohanjha123 Is this the expected behavior for the function? If yes - Function seems to behave in the expected way in Data100 hub while showing different behavior in datahub. 
![image](https://user-images.githubusercontent.com/2306166/210826686-c44faccf-ed60-43df-8016-d8355d55ccca.png)

The delta I see between Data 100 and Datahub image is the ipywidgets package version. Data 100 hub has v8.1.0 (thanks to this PR - https://github.com/berkeley-dsep-infra/datahub/pull/3608/commits/355b3de7e5faaa3cfc063b13597b1dfa9100d31a which was based on the fall package refresh requested here -  https://github.com/berkeley-dsep-infra/datahub/issues/3605) and datahub has version v7.7.2 (https://github.com/berkeley-dsep-infra/datahub/blob/staging/deployments/datahub/images/default/infra-requirements.txt). 

If my understanding is right, we need to upgrade package version of ipywidgets in Datahub image to mimic the requested behavior. Thoughts?

**Comment by felder** [comment]
@balajialg did you mean 8.0.1?

I'm also confused, because I see no evidence that data100 is using that.

![Screen Shot 2023-01-05 at 10 23 44 AM](https://user-images.githubusercontent.com/1211730/210829867-3f09141b-af67-4755-82e1-308b9bc2afe8.png)

https://github.com/berkeley-dsep-infra/datahub/blob/efc6b040174d937ee7e0c596ad488c27390fd7fe/deployments/stat159/image/infra-requirements.txt#L19

(note data100 shares its image with stat159)

What am I missing?

**Comment by balajialg** [comment]
@felder Thanks for pointing it out! Yes - I meant v8.0.1. @yuvipanda reverted ipywidgets to the older version in the Stat 159 image due to dependency hell with qgrid which I missed reviewing while commenting earlier - https://github.com/berkeley-dsep-infra/datahub/pull/3620/commits/6258a9a97518a21fb5eb0048977bcc3ea92a2b68. 

Now, its confusing why two different hubs have different behavior for the same version of the package.

**Comment by rohanjha123** [comment]
The earlier photo posted by @balajialg is indeed the expected functionality - thanks so much! I guess it boils down to this:  I was in datahub building content for the new class econ 148. Econ 148 will need to be put on a hub, as mentioned here: https://github.com/berkeley-dsep-infra/datahub/issues/4005. So, I guess I don't mind if the interact feature doesn't work in datahub as long as it works in whichever hub econ 148 ends up using. Thanks so much for all the help with this @balajialg @felder, I really appreciate it!

**Comment by felder** [comment]
Ok so comparing conda installed packages between datahub and data100 (where this works) I was able to narrow this down to ipykernel

On datahub ipykernel==6.19.3 on data100 ipykernel==6.6.0

After giving myself enough ram on datahub for my kernel not to die...

Running mamba install ipykernel==6.6.0 via the terminal and then restarting my kernel fixes this

Proposed fix would be to add ipykernel==6.6.0 to the dependency list in environment.yml

However I'm curious how ipykernel gets pulled in, because it's not explicitly listed in any of the dependency lists.  Makes me worried about what downgrading it might break.

**Comment by felder** [comment]
Upgrading ipykernel to 6.19.4 also fixes!  That seems like a better thing to do IMO.

**Comment by felder** [comment]
Ok this is on its way to staging.  It'll be 1.5hrs or so before it lands there.  I'll test tomorrow and if it workie, it'll be a short trip over to prod and this will be fixed.

**Comment by felder** [comment]
Just leaving a final comment here.

This was merged to prod in #4086 and I verified that it is working in prod.
